File: src/StayFit/Models/Avaliacao.cs
using System.ComponentModel.DataAnnotations;

namespace StayFit.Models
{
    public class Avaliacao
    {
        public Avaliacao()
        {
            IsPraticante = false;
            DataAvaliacao = DateTime.Now;
            Objetivos = "Emagrecer";
            Peso = 0;
            Altura = 0;
        }

        [Key]
        public int AvaliacaoId { get; set; }

        public DateTime DataAvaliacao { get; set; }
        public string Objetivos { get; set; }


        [Required(ErrorMessage = "Informe o Peso Corporal do Cliente")]
        public float Peso { get; set; }

        [Required(ErrorMessage = "Informe a Altura do Cliente")]
        public float Altura { get; set; }
        public Boolean? IsPraticante { get; set; }
        public bool? IsTomaMedicamentos { get; set; }
        public String? Medicamentos { get; set; }
        public bool? IsFumante { get; set; }
        public bool? IsProblemaSaude{ get; set; }
        public String? ProblemaSaude { get; set; }      
        public bool? IsDeficiente { get; set; }
        public String? Deficiencia { get; set; }
        public String? Observacoes { get; set; }


        public float? CircunferenciaBracoDir { get; set; }
        public float? CircunferenciaBracoEsq { get; set; }
        public float? CircunferenciaAnteBracoDir { get; set; }
        public float? CircunferenciaAnteBracoEsq { get; set; }
        public float? CircunferenciaAbdomen { get; set; }
        public float? CircunferenciaQuadril { get; set; }
        public float? CircunferenciaCintura { get; set; }
        public float? CircunferenciaCoxaEsq { get; set; }
        public float? CircunferenciaCoxaDir { get; set; }
        public float? CircunferenciaPernaEsq { get; set; }
        public float? CircunferenciaPernaDir { get; set; }

        public float? PercentualGordura { get; set; }
        public float? MassaMagra { get; set; }
        public float? MassaGorda { get; set; }
        public float? MassaMuscular { get; set; }
        public float? MassaOssea { get; set; }
        public float? MassaResidual { get; set; }

        public int? FrecCardRepouso { get; set; }
        public int? FrecCardM1 { get; set; }
        public int? FrecCardM2 { get; set; }
        public int? FrecCardM3 { get; set; }
        public int? FrecCardMaxima { get; set; }

        public float? VoMax { get; set; }


        public float? TempoEsteira { get; set; }
    
    }
}

File: src/StayFit/wwwroot/js/Avaliacao.js
﻿ 
function updateAnamnese() {
   
    let Altura = document.getElementById("Altura").value;
    let Peso = document.getElementById("Peso").value;    
    let IsTomaMedicamentos = document.getElementById("IsTomaMedicamentos");
    let objetivos = document.getElementById("Objetivos").value;
    let medicamentos = document.getElementById("Medicamentos").value;
    let observacoes = document.getElementById("Observacoes").value;
    let deficiencia = document.getElementById("Deficiencia").value;
    let isProblemaSaude = document.getElementById("IsProblemaSaude");
    let isDeficiente = document.getElementById("IsDeficiente");
    let isFumante = document.getElementById("IsFumante");
    let isPraticante = document.getElementById("IsPraticante");
    let problemaSaude = document.getElementById("ProblemaSaude").value;

    let isMed = isTomaMedicamentos.options[isTomaMedicamentos.selectedIndex].value;
    let isFum = isFumante.options[isFumante.selectedIndex].value;
    let isDef = isDeficiente.options[isDeficiente.selectedIndex].value;
    let isPrat = isPraticante.options[isPraticante.selectedIndex].value;
    let isPSau = isProblemaSaude.options[isProblemaSaude.selectedIndex].value;
     
    let avaliacao = {
        "objetivos": objetivos,
        "peso": Peso,
        "altura": Altura,
        "isPraticante": isPrat,
        "isTomaMedicamentos": 0,
        "medicamentos": medicamentos,
        "isFumante": 1,
        "isProblemaSaude": 0,
        "problemaSaude": problemaSaude,
        "isDeficiente": isDef,
        "deficiencia": deficiencia,
        "observacoes": observacoes,

    };


    //let avalia = {
    //    "objetivos": "TESTE",
    //    "peso": 150,
    //    "altura": 190,
    //    "isPraticante": false,
    //    "isTomaMedicamentos": null,
    //    "medicamentos": null,
    //    "isFumante": null,
    //    "isProblemaSaude": null,
    //    "problemaSaude": null,
    //    "isDeficiente": null,
    //    "deficiencia": null,
    //    "observacoes": null,
    //    "circunferenciaBracoDir": null,
    //    "circunferenciaBracoEsq": null,
    //    "circunferenciaAnteBracoDir": null,
    //    "circunferenciaAnteBracoEsq": null,
    //    "circunferenciaAbdomen": null,
    //    "circunferenciaQuadril": null,
    //    "circunferenciaCintura": null,
    //    "circunferenciaCoxaEsq": null,
    //    "circunferenciaCoxaDir": null,
    //    "circunferenciaPernaEsq": null,
    //    "circunferenciaPernaDir": null,
    //    "percentualGordura": null,
    //    "massaMagra": null,
    //    "massaGorda": null,
    //    "massaMuscular": null,
    //    "massaOssea": null,
    //    "massaResidual": null,
    //    "frecCardRepouso": null,
    //    "frecCardM1": null,
    //    "frecCardM2": null,
    //    "frecCardM3": null,
    //    "frecCardMaxima": null,
    //    "voMax": null,
    //    "tempoEsteira": null

    //};
  
    console.log(avaliacao);

    fetch('/Instrutor/UpdateAnamnese', {
        method: 'POST',
        headers: {
            'Content-type': 'application/json',
        },
        body: JSON.stringify(avaliacao)
    }).then(function (response) {
        if (response.ok) {
            alert("teste " + response);
            alert(isFumante);
            return response.json();
        }
        return Promise.reject(response);
    }).then(data => {
        let dados = JSON.parse(data);
        console.log('Success:', dados);
        if (data) {
            
            //window.location.replace('http://localhost:5247/Instrutor/Av');
        }
    }).catch(function (error) {
        console.warn(error);
    });

}

function ListarTreino() {
    let place_treinos = '';
    let data = JSON.parse(localStorage.getItem("TreinosLocal"));
    let dataEx = JSON.parse(localStorage.getItem("NomeExercicios"));
    if (data !== null) {
        data.forEach((treino, index) => {
            place_treinos += ` 
            <div>
                <h4><span>${dataEx[index]} </span></h4>
                <h5><span>Repeticoes: ${treino.repetitionNumber} </span><span>Series: ${treino.series}</span> </h5>
            </div>`
        });
    }

    document.querySelector('.treinos-card').innerHTML = place_treinos;
    place_treinos = ''
}